x-pack/metricbeat/module/statsd/server/data_test.go

```diff
@@ -5,17 +5,844 @@
+			metricName:  "smart_sensor_operator.poked_tasks",
+			metricValue: gaugeValue,
+			expected: common.MapStr{
+				"smart_sensor_operator_poked_tasks": gaugeValue,
+			},
+		},
+		{
+			metricName:  "smart_sensor_operator.poked_success",
+			metricValue: gaugeValue,
+			expected: common.MapStr{
+				"smart_sensor_operator_poked_success": gaugeValue,
+			},
+		},
+		{
+			metricName:  "smart_sensor_operator.poked_exception",
+			metricValue: gaugeValue,
+			expected: common.MapStr{
+				"smart_sensor_operator_poked_exception": gaugeValue,
+			},
+		},
+		{
+			metricName:  "smart_sensor_operator.exception_failures",
+			metricValue: gaugeValue,
+			expected: common.MapStr{
+				"smart_sensor_operator_exception_failures": gaugeValue,
+			},
+		},
+		{
+			metricName:  "smart_sensor_operator.infra_failures",
+			metricValue: gaugeValue,
+			expected: common.MapStr{
+				"smart_sensor_operator_infra_failures": gaugeValue,
+			},
+		},
+		{
+			metricName:  "dagrun.dependency-check.a_dag_id",
+			metricValue: timerValue,
+			expected: common.MapStr{
+				"dag_id":               "a_dag_id",
+				"dag_dependency_check": timerValue,
+			},
+		},
+		{
+			metricName:  "dag.a_dag_id.a_task_id.duration",
+			metricValue: timerValue,
+			expected: common.MapStr{
+				"dag_id":        "a_dag_id",
+				"task_id":       "a_task_id",
+				"task_duration": timerValue,
+			},
+		},
+		{
+			metricName:  "dag_processing.last_duration.a_dag_file",
+			metricValue: timerValue,
+			expected: common.MapStr{
+				"dag_file":          "a_dag_file",
+				"dag_last_duration": timerValue,
+			},
+		},
+		{
+			metricName:  "dagrun.duration.success.a_dag_id",
+			metricValue: timerValue,
+			expected: common.MapStr{
+				"dag_id":               "a_dag_id",
+				"success_dag_duration": timerValue,
+			},
+		},
+		{
+			metricName:  "dagrun.duration.failed.a_dag_id",
+			metricValue: timerValue,
+			expected: common.MapStr{
+				"dag_id":              "a_dag_id",
+				"failed_dag_duration": timerValue,
+			},
+		},
+		{
+			metricName:  "dagrun.schedule_delay.a_dag_id",
+			metricValue: timerValue,
+			expected: common.MapStr{
+				"dag_id":             "a_dag_id",
+				"dag_schedule_delay": timerValue,
+			},
+		},
+		{
+			metricName:  "scheduler.critical_section_duration",
+			metricValue: timerValue,
+			expected: common.MapStr{
+				"scheduler_critical_section_duration": timerValue,
+			},
+		},
+		{
+			metricName:  "dagrun.a_dag_id.first_task_scheduling_delay",
+			metricValue: timerValue,
+			expected: common.MapStr{
+				"dag_id":                          "a_dag_id",
+				"dag_first_task_scheduling_delay": timerValue,
+			},
+		},
+		{
+			metricName:  "not_mapped_metric",
+			metricValue: timerValue,
+			expected:    common.MapStr{},
+		},
+	} {
+		metricSetFields := common.MapStr{}
+		builtMappings, _ := buildMappings(mappings)
+		eventMapping(test.metricName, test.metricValue, metricSetFields, builtMappings)
+
+		assert.Equal(t, test.expected, metricSetFields)
```

Comment:
Consider adding some message that helps to identify what case failed in case of failure. Or run each iteration inside a different subtest.